tests/fast-integration/installer/helm.js

```diff
@@ -0,0 +1,381 @@
+const execa = require("execa");
+const k8s = require("@kubernetes/client-node");
+const { join } = require("path");
+const { installIstio, upgradeIstio } = require("./istioctl");
+const pRetry = require("p-retry");
+
+const notDeployed = "not-deployed";
+
+async function waitForNodesToBeReady(timeout = "180s") {
+  await execa("kubectl", [
+    "wait",
+    "--for=condition=Ready",
+    "nodes",
+    "--all",
+    `--timeout=${timeout}`,
+  ]);
+  console.log("All nodes are ready!");
+}
+
+// TODO: replace by kubectlApply from utils
+async function kubectlApply(path, namespace) {
+  const args = ["apply", "-f", path];
+  if (!!namespace) {
+    args.push("-n", namespace);
+  }
+  await execa("kubectl", args);
+}
+
+async function updateCoreDNSConfigMap() {
+  const kc = new k8s.KubeConfig();
+  kc.loadFromDefault();
+
+  const cmName = "coredns";
+  const cmNamespace = "kube-system";
+
+  const k8sCoreV1Api = kc.makeApiClient(k8s.CoreV1Api);
+  const { body } = await k8sCoreV1Api.readNamespacedConfigMap(
+    "coredns",
+    "kube-system"
+  );
+
+  const { stdout: registryIP } = await execa("docker", [
+    "inspect",
+    "-f",
+    "{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}",
+    "/registry.localhost",
+  ]);
+
+  // this file need to be updated when we update k3s to use k8s 1.20
+  body.data["Corefile"] = `registry.localhost:53 {
+    hosts {
+        ${registryIP} registry.localhost
+    }
+}
+.:53 {
+    errors
+    health
+    rewrite name regex (.*)\.local\.kyma\.dev istio-ingressgateway.istio-system.svc.cluster.local
+    ready
+    kubernetes cluster.local in-addr.arpa ip6.arpa {
+      pods insecure
+      upstream
+      fallthrough in-addr.arpa ip6.arpa
+    }
+    hosts /etc/coredns/NodeHosts {
+      reload 1s
+      fallthrough
+    }
+    prometheus :9153
+    forward . /etc/resolv.conf
+    cache 30
+    loop
+    reload
+    loadbalance
+}
+`;
+  await k8sCoreV1Api.replaceNamespacedConfigMap(cmName, cmNamespace, body);
+}
+
+async function removeKymaGatewayCertsYaml(pathToResources) {
+  // TODO: find some other mechanism, deleting that file is destructive
+  // we can copy resources directory so that we can act on it
+  try {
+    await execa("rm", [
+      join(
+        pathToResources,
+        "core/charts/gateway/templates/kyma-gateway-certs.yaml"
+      ),
+    ]);
+  } catch (err) {
+    console.log("kyma-gateway-certs.yaml already deleted");
+  }
+}
+
+async function helmStatus(release, namespace) {
+  try {
+    const { stdout } = await execa("helm", [
+      "status",
+      release,
+      "-n",
+      namespace,
+      "-ojson",
+    ]);
+    return JSON.parse(stdout).info.status;
+  } catch (err) {
+    return notDeployed;
+  }
+}
+
+async function helmUninstall(release, namespace) {
+  await execa("helm", ["uninstall", release, "-n", namespace]);
+}
+
+async function helmInstallUpgrade(
+  release,
+  chart,
+  namespace,
+  setValues,
```

Comment:
Please rename `setValues` to just `values`. `set...` may suggest it is a function.